lib/jekyll-titles-from-headings/generator.rb

```diff
@@ -112,6 +112,13 @@ def inferred_title?(document)
 
     def strip_title!(document)
       document.content.gsub!(TITLE_REGEX, "")
+      document.content.strip!
+
+      if document.is_a?(Jekyll::Document) &&
```

Comment:
I'd move this logic into a `set_excerpt?` or similar helper method to keep things readable.

Reply:
you're right. I'll update the PR